src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -808,12 +821,9 @@ and the second `setWidth` is a non-trivial extension.
 -- `simp` can discharge the side condition itself.
 @[simp] theorem setWidth_setWidth {x : BitVec u} {w v : Nat} (h : ¬ (v < u ∧ v < w)) :
     setWidth w (setWidth v x) = setWidth w x := by
-  ext
-  simp_all only [getLsbD_setWidth, decide_true, Bool.true_and, Bool.and_iff_right_iff_imp,
-    decide_eq_true_eq]
-  intro h
-  replace h := lt_of_getLsbD h
-  omega
+  ext i ih
+  have h := @lt_of_getLsbD u x i
+  by_cases h : x.getLsbD i = true <;> simp [h] at * <;> omega
```

Comment:
Strange that both are called `h`. Drop one name?

Reply:
Addressed.